### PR TITLE
Bugfix/PATRO23-314-323 - fix invalid transaction cancel

### PR DESCRIPTION
**Pull request by DawidBski**
### PATRO23-314 - add permission checking
https://tracker.intive.com/jira/browse/PATRO23-314

### PATRO23-323 - fix invalid transaction cancel endpoint and its validation
https://tracker.intive.com/jira/browse/PATRO23-323

**Added incidentally:**

- Add ProducesResponseType attribute to RemoveBudget and CancelBudgetTransaction actions

### CODE REVIEW

File: src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs [hunk 1]
```diff
@@ -10,26 +10,31 @@ namespace Intive.Patronage2023.Modules.Budget.Application.Budget.CancelBudgetTra
...
 			.NotNull();
 
-		this.RuleFor(transaction => transaction.Id)
-			.MustAsync(async (x, cancellation) => await this.IsTransactionExists(x));
+		this.RuleFor(transaction => transaction.TransactionId)
+			.NotEmpty()
+			.NotNull();
+
+		this.RuleFor(transaction => transaction)
+			.MustAsync(async (x, cancellation) => await this.IsTransactionBelongingToBudget(x.BudgetId, x.TransactionId));
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs:32]: Leave some error message

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs (hunk 1 above)
Comment (DawidBski) [reply src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs]: Error message and new rules are added.

**Review by pzajaczkowski — APPROVED** [review]

**Review by Kepeczsz — APPROVED** [review]

File: src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs [hunk 2]
```diff
@@ -10,26 +10,55 @@ namespace Intive.Patronage2023.Modules.Budget.Application.Budget.CancelBudgetTra
...
 			.NotNull();
 
-		this.RuleFor(transaction => transaction.Id)
-			.MustAsync(async (x, cancellation) => await this.IsTransactionExists(x));
+		this.RuleFor(transaction => transaction.TransactionId)
+			.NotEmpty()
+			.NotNull();
+
+		this.RuleFor(transaction => transaction.TransactionId)
+			.MustAsync(async (x, cancellation) => await this.IsTransactionExists(x))
```
Comment (mateuszmuszynski) [inline src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs:35]: Some say that's a good practice to start bool returning methods with **Is** suffix, but these method names are not gramatically correct.
Imo methods that returns a boolean value should be named as a yes-no question.
Example:
IsTransactionExists -> TransactionExists / ShouldExist

Comment (mateuszmuszynski) [reply src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs:35]: Please also update other rules

**Review by mateuszmuszynski — CHANGES_REQUESTED** [review]

File: src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs (hunk 2 above)
Comment (DawidBski) [reply src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs:35]: What do you mean other rules? Only in my validator or rather in every?

Comment (DawidBski) [reply src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs]: Ok Done.

**Review by mateuszmuszynski — APPROVED** [review]

File: src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs (hunk 2 above)
Comment (mateuszgiedzinski) [reply src/modules/budget/application/Budget/CancelBudgetTransaction/CancelBudgetTransactionValidator.cs]: For me both forms are correct. I disagree that our code should be gramatically correct. It must be consistent so if all our method which check condition has "Is", "Has" or "Can" prefixes we should follow that rule.

**Review by mateuszgiedzinski — APPROVED** [review]

**Pull request merged**